resources/assets/lib/beatmap-discussions/editor-beatmap-selector.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import { BeatmapIcon } from 'beatmap-icon';
+import * as React from 'react';
+import { Node, Transforms } from 'slate';
+import { ReactEditor } from 'slate-react';
+import IconDropdownMenu, { MenuItem } from './icon-dropdown-menu';
+import { SlateContext } from './slate-context';
+
+interface Props {
+  beatmaps: Beatmap[];
+  disabled: boolean;
+  element: Node;
+}
+
+export default class EditorBeatmapSelector extends React.Component<Props> {
+  static contextType = SlateContext;
+  menuOptions: MenuItem[];
+
+  constructor(props: Props) {
+    super(props);
+
+    this.menuOptions = [];
+    this.menuOptions.push({
+      icon: <i className='fas fa-fw fa-star-of-life beatmap-discussion-editor__menu-icon' />,
+      id: 'all',
+      label: osu.trans('beatmaps.discussions.mode.scopes.generalAll'),
+    });
+
+    this.props.beatmaps.forEach((beatmap: Beatmap) => {
+      if (beatmap.deleted_at) {
+        return;
+      }
+
+      this.menuOptions.push({
```

Comment:
the props might never be changed but it sure is kind of weird seeing this being permanently set based on initial props

Reply:
Yeah not sure, the idea was to save regenerating the menu on every render... probably fine just moving it to render? ...but then again the same time the page is already pretty heavy, so 🤷‍♂️